src/main/electron-app/runnables/setup-ipc-main-handlers/setup-ipc-main-handlers.ts

```diff
@@ -78,11 +82,11 @@ export const setupIpcMainHandlers = ({
   });
 
   ipcMainHandle(clusterKubectlApplyAllHandler, async (event, clusterId: ClusterId, resources: string[], extraArgs: string[]) => {
-    appEventBus.emit({ name: "cluster", action: "kubectl-apply-all" });
+    emitAppEvent({ name: "cluster", action: "kubectl-apply-all" });
     const cluster = ClusterStore.getInstance().getById(clusterId);
 
     if (cluster) {
-      const applier = new ResourceApplier(cluster);
+      const applier = createResourceApplier(cluster);
```

Comment:
As strategy, it could be better to not inject factory functions, if an already "factored" instance can be injected instead. Here: `applier` could be the dependency instead of `createResourceApplier`.

Reply:
small difference

Reply:
Sometimes, sometimes not :) Resolved for already communicating the general point somewhere else.